std/zig/parse.zig

```diff
@@ -1,3659 +1,3044 @@
-                    try stack.append(State{ .ExpectToken = Token.Id.LParen });
-                    continue;
-                }
+        if (eatToken(it, .Semicolon)) |_| return node;
+        if (try parseBlock(arena, it, tree)) |body_node| {
+            fn_node.body_node = body_node;
+            return node;
+        }
 
-                prevToken(&tok_it, tree);
-                continue;
-            },
-            State.VarDeclSection => |var_decl| {
-                try stack.append(State{ .VarDeclEq = var_decl });
-
-                const next_token = nextToken(&tok_it, tree);
-                const next_token_index = next_token.index;
-                const next_token_ptr = next_token.ptr;
-                if (next_token_ptr.id == Token.Id.Keyword_linksection) {
-                    try stack.append(State{ .ExpectToken = Token.Id.RParen });
-                    try stack.append(State{ .Expression = OptionalCtx{ .RequiredNull = &var_decl.section_node } });
-                    try stack.append(State{ .ExpectToken = Token.Id.LParen });
-                    continue;
-                }
+        try tree.errors.push(AstError{
+            .ExpectedSemiOrLBrace = AstError.ExpectedSemiOrLBrace{ .token = it.index },
+        });
+        return null;
+    }
 
-                prevToken(&tok_it, tree);
-                continue;
-            },
-            State.VarDeclEq => |var_decl| {
-                const token = nextToken(&tok_it, tree);
-                const token_index = token.index;
-                const token_ptr = token.ptr;
-                switch (token_ptr.id) {
-                    Token.Id.Equal => {
-                        var_decl.eq_token = token_index;
-                        stack.append(State{ .VarDeclSemiColon = var_decl }) catch unreachable;
-                        try stack.append(State{ .Expression = OptionalCtx{ .RequiredNull = &var_decl.init_node } });
-                        continue;
-                    },
-                    Token.Id.Semicolon => {
-                        var_decl.semicolon_token = token_index;
-                        continue;
-                    },
-                    else => {
-                        ((try tree.errors.addOne())).* = Error{ .ExpectedEqOrSemi = Error.ExpectedEqOrSemi{ .token = token_index } };
-                        return tree;
-                    },
-                }
-            },
+    if (extern_export_token) |token| {
+        if (tree.tokens.at(token).id == .Keyword_inline) {
+            _ = prevToken(it); // rewind "inline"
+            return null;
+        }
+    }
 
-            State.VarDeclSemiColon => |var_decl| {
-                const semicolon_token = nextToken(&tok_it, tree);
+    const thread_local_token = eatToken(it, .Keyword_threadlocal);
+
+    if (try parseVarDecl(arena, it, tree)) |node| {
+        var var_decl = node.cast(Node.VarDecl).?;
+        var_decl.*.doc_comments = doc_comments;
```

Comment:
A few `ptr.*.field`-style accesses pepper this PR, where otherwise it uses the `ptr.field` implicit dereferencing. I wasn't sure which is more idiomatic (I'm guessing implicit); it should probably be made consistent.

Reply:
we should probably, put something about that in the style guide.

Reply:
:+1: I'll not worry about it for now and wait until there is a style guideline for it.